lib/active_merchant/billing/gateways/merchant_e_solutions.rb

```diff
@@ -31,6 +31,9 @@ def authorize(money, creditcard_or_card_id, options = {})
         post[:moto_ecommerce_ind] = options[:moto_ecommerce_ind] if options[:moto_ecommerce_ind]
         post[:recurring_pmt_num] = options[:recurring_pmt_num] if options[:recurring_pmt_num]
         post[:recurring_pmt_count] = options[:recurring_pmt_count] if options[:recurring_pmt_count]
+        post[:card_on_file] = options[:card_on_file] if options[:card_on_file]
+        post[:cit_mit_indicator] = options[:cit_mit_indicator] if options[:cit_mit_indicator]
+        post[:account_data_source] = options[:account_data_source] if options[:account_data_source]
```

Comment:
This all looks good, but I think this is looking a bit bloated. We should create a common method to append these values to the `post` body and call that method within the `authorize` and `purchase` method transactions.

Reply:
Since these all look like some variation of stored credentials, we should go ahead and make a method for `add_stored_credentials` and then move these and the `recurring_*` fields as well.

Reply:
Done